client/internal/receivedprocessor.go

```diff
@@ -209,6 +210,10 @@ func (r *receivedProcessor) rcvOpampConnectionSettings(ctx context.Context, sett
 		return
 	}
 
+	if r.hasCapability(protobufs.AgentCapabilities_AgentCapabilities_ReportsHeartbeat) {
+		r.sender.SetHeartbeatInterval(time.Duration(settings.Opamp.HeartbeatIntervalSeconds) * time.Second)
```

Comment:
Once the error is returned by `SetHeartbeatInterval` we can probably also log the error here to indicate the setting was ignored.